### PR TITLE
Tests failing due to mysql/mariadb errors

### PR DESCRIPTION
**Pull request by rv0lt**
## Read this before submitting the PR

1. Always create a Draft PR first
2. Go through sections 1-5 below, fill them in and check all the boxes
3. Make sure that the branch is updated; if there's an "Update branch" button at the bottom of the PR, rebase or update branch.
4. When all boxes are checked, information is filled in, and the branch is updated: mark as Ready For Review and tag reviewers (top right)
5. Once there is a submitted review, implement the suggestions (if reasonable, otherwise discuss) and request an new review.

If there is a field which you are unsure about, enter the edit mode of this description or go to the [PR template](../.github/pull_request_template.md); There are invisible comments providing descriptions which may be of help.

## 1. Description / Summary

Since 10/12 running the tests results in error in connectivity between pytests and the database

dds_database     | 2024-12-11 13:13:24 46 [Warning] Aborted connection 46 to db: 'DeliverySystemTest' user: 'root' host: '172.20.0.4' (Got an error reading communication packets)

 

dds_backend      | E       sqlalchemy.exc.ProgrammingError: (pymysql.err.ProgrammingError) (1146, "Table 'DeliverySystemTest.projects' doesn't exist") 

After some debugging, found that the error is in the dumping between the data from DeliverySystemTestBase to DeliverySystemTest. Essentially, this instruction generated by the dump is not able to be processed by mariadb:

-------------- /*M!100616 SET @OLD_NOTE_VERBOSITY=@@NOTE_VERBOSITY, NOTE_VERBOSITY=0 */ -------------- ERROR 1193 (HY000) at line 12: Unknown system variable 'NOTE_VERBOSITY' CompletedProcess(args=['mariadb', '-h', 'db', '-u', 'root', '-pTEST_PASSWORD', '--skip_ssl', 'DeliverySystemTest'], returncode=1)

## 2. Jira task / GitHub issue

[_Link to the github issue or add the Jira task ID here._](https://scilifelab.atlassian.net/jira/software/projects/HMS/boards/13?selectedIssue=HMS-2203)

## 3. Type of change

What _type of change(s)_ does the PR contain?

**Check the relevant boxes below. For an explanation of the different sections, enter edit mode of this PR description template.**

- [X] New feature
  - [ ] Breaking: _Why / How? Add info here._ 
  - [ ] Non-breaking 
- [ ] Database change: _Remember the to include a new migration version, **or** explain here why it's not needed._ 
- [ ] Bug fix 
- [ ] Security Alert fix 
  - [ ] Package update 
    - [ ] Major version update 
- [ ] Documentation 
- [ ] Workflow 
- [ ] Tests **only** 

## 4. Additional information

- [x] [Sprintlog](../SPRINTLOG.md) 
- [ ] Blocking PRs 
  - [ ] Merged 
- [ ] PR to `master` branch: \_If checked, read [the release instructions](../doc/procedures/new_release.md) 
  - [ ] I have followed steps 1-8. 

## 5. Actions / Scans

_Check the boxes when the specified checks have passed._

**For information on what the different checks do and how to fix it if they're failing, enter edit mode of this description or go to the [PR template](../.github/pull_request_template.md).**

- [x] **Black**

- [x] **Prettier**

- [x] **Yamllint**

- [x] **Tests**

- [x] **CodeQL**

- [x] **Trivy**

- [x] **Snyk**

### CODE REVIEW

**Comment by rv0lt** [comment]
Refer to https://scilifelab.atlassian.net/jira/software/projects/HMS/boards/13?selectedIssue=HMS-2203 for a discussion on alternatives

File: SPRINTLOG.md [hunk 1]
```diff
@@ -464,3 +464,4 @@ _Nothing merged during this sprint_
 - Change the error raised upon attempt to download data after a password reset to an AuthenticationError to avoid getting an alert ([#1571](https://github.com/ScilifelabDataCentre/dds_web/pull/1571))
 - Filter out the MaintenanceModeException from the logs ([#1573](https://github.com/ScilifelabDataCentre/dds_web/pull/1573))
 - Bugfix: Quick and dirty change to prevent `dds ls --tree` from failing systematically ([#1575](https://github.com/ScilifelabDataCentre/dds_web/pull/1575)
+- update backend Dockerfile to pin a fixed version of mariadb-client ([#1581](https://github.com/ScilifelabDataCentre/dds_web/pull/1581)
```
Comment (i-oden) [inline SPRINTLOG.md:467]: Just a tiny typo
```suggestion
- Update backend Dockerfile to pin a fixed version of mariadb-client ([#1581](https://github.com/ScilifelabDataCentre/dds_web/pull/1581)
```

File: tests/conftest.py [hunk 2]
```diff
@@ -103,9 +103,25 @@ def fill_basic_db(db):
 def new_test_db(uri):
     dbname = uri[uri.rindex("/") + 1 :]
     dbname_base = DATABASE_URI_BASE[DATABASE_URI_BASE.rindex("/") + 1 :]
-    dump_args = ["mysqldump", "-h", "db", "-u", "root", f"-p{mysql_root_password}", dbname_base]
-    load_args = ["mysql", "-h", "db", "-u", "root", f"-p{mysql_root_password}", dbname]
-    proc1 = subprocess.run(dump_args, capture_output=True)
```
Comment (i-oden) [inline tests/conftest.py:108]: This row is deleted. Can you add info here about why if this was intentional?

Comment (rv0lt) [reply tests/conftest.py:108]: It wasn't doing anything. My best guess is that it was a testing thing that was left there.

It's the next subprocess which pipes the output to be read by the load

**Comment by i-oden** [comment]
I'm getting loads of these errors when I run the tests locally: 
```bash
dds_database     | 2024-12-12 10:24:56 49 [ERROR] Incorrect definition of table mysql.column_stats: expected column 'hist_type' at position 9 to have type enum('SINGLE_PREC_HB','DOUBLE_PREC_HB','JSON_HB'), found type enum('SINGLE_PREC_HB','DOUBLE_PREC_HB').
```

**Comment by rv0lt** [comment]
Interactive or locally? I haven't seen it

> I'm getting loads of these errors when I run the tests locally:
> 
> ```shell
> dds_database     | 2024-12-12 10:24:56 49 [ERROR] Incorrect definition of table mysql.column_stats: expected column 'hist_type' at position 9 to have type enum('SINGLE_PREC_HB','DOUBLE_PREC_HB','JSON_HB'), found type enum('SINGLE_PREC_HB','DOUBLE_PREC_HB').
> ```

**Comment by i-oden** [comment]
> Interactive or locally? I haven't seen it
> 
> > I'm getting loads of these errors when I run the tests locally:
> > ```shell
> > dds_database     | 2024-12-12 10:24:56 49 [ERROR] Incorrect definition of table mysql.column_stats: expected column 'hist_type' at position 9 to have type enum('SINGLE_PREC_HB','DOUBLE_PREC_HB','JSON_HB'), found type enum('SINGLE_PREC_HB','DOUBLE_PREC_HB').
> > ```

With the one-line command (`docker-compose -f docker-compose.yml -f tests/docker-compose-test.yml up --build --exit-code-from backend`), not the interactive one (because atm the interactive one is not letting me enter the container to run pytest at all)

**Comment by i-oden** [comment]
@ScilifelabDataCentre/teamhermes Also:
```
Code scanning results / Trivy Completed in 4s — 2 configurations not found 
```

**Comment by i-oden** [comment]
> I'm getting loads of these errors when I run the tests locally:
> 
> ```shell
> dds_database     | 2024-12-12 10:24:56 49 [ERROR] Incorrect definition of table mysql.column_stats: expected column 'hist_type' at position 9 to have type enum('SINGLE_PREC_HB','DOUBLE_PREC_HB','JSON_HB'), found type enum('SINGLE_PREC_HB','DOUBLE_PREC_HB').
> dds_database     | 2024-12-12 10:24:56 49 [ERROR] Incorrect definition of table mysql.column_stats: expected column 'histogram' at position 10 to have type longblob, found type varbinary(255).
> ```

@ScilifelabDataCentre/teamhermes does anyone else get these errors when running locally with the command I wrote here (and that is in the readme)?  https://github.com/ScilifelabDataCentre/dds_web/pull/1581#issuecomment-2538493403

**Comment by rv0lt** [comment]
It seems that this is a global issue with the action, I peeked and the others PR seems to be the same

> @ScilifelabDataCentre/teamhermes Also:
> 
> ```
> Code scanning results / Trivy Completed in 4s — 2 configurations not found 
> ```

**Comment by rv0lt** [comment]
Could you also do 
`docker-compose down --rmi all -v --remove-orphans`

And then run the tests again, to be sure?

> ```shell
> Incorrect definition of table mysql.column_stats: expected column 'histogram' at position 10 to have type longblob, found type varbinary(255).
> ```

**Comment by i-oden** [comment]
> Could you also do `docker-compose down --rmi all -v --remove-orphans`
> 
> And then run the tests again, to be sure?
> 
> > ```shell
> > Incorrect definition of table mysql.column_stats: expected column 'histogram' at position 10 to have type longblob, found type varbinary(255).
> > ```

Yeah I'm not getting the errors anymore, but I am getting that all the tests fail instead.

**Comment by rv0lt** [comment]
Is the database container logging something like?:

Aborted connection 46 to db: 'DeliverySystemTest' user: 'root' host: '172.20.0.4' (Got an error reading communication packets)

Because that was the error that was happening

> > Could you also do `docker-compose down --rmi all -v --remove-orphans`
> > And then run the tests again, to be sure?
> > > ```shell
> > > Incorrect definition of table mysql.column_stats: expected column 'histogram' at position 10 to have type longblob, found type varbinary(255).
> > > ```
> 
> Yeah I'm not getting the errors anymore, but I am getting that all the tests fail instead. With `E`

**Comment by i-oden** [comment]
> Is the database container logging something like?:
> 
> Aborted connection 46 to db: 'DeliverySystemTest' user: 'root' host: '172.20.0.4' (Got an error reading communication packets)
> 
> Because that was the error that was happening
> 
> > > Could you also do `docker-compose down --rmi all -v --remove-orphans`
> > > And then run the tests again, to be sure?
> > > > ```shell
> > > > Incorrect definition of table mysql.column_stats: expected column 'histogram' at position 10 to have type longblob, found type varbinary(255).
> > > > ```
> > 
> > 
> > Yeah I'm not getting the errors anymore, but I am getting that all the tests fail instead. With `E`

It's doing stuff now and there are no errors anywhere. I probably made a mistake somewhere, like running `docker-compose` instead of `docker compose` but anyway, all good now. Going to approve now 👍🏻

**Review by i-oden — APPROVED** [review]
All good 👍🏻  You can merge and then check in the staging or dev instance that everything is ok. You can also do a redeployment in test just to check so that we can do the release stuff asap.

**Pull request merged**